### ISSUE TITLE
Feature: Ability to add text to A element on wp-block-post-featured-image

### ISSUE THREAD

**Issue by blueocto** [WordPress/gutenberg#41077]
## What problem does this address?
Currently, when you opt to add a link around a Featured Image, inside the Query Loop block, it does not apply a name/title attribute. This is currently flagging in Google Lighthouse/Core Web Vitals as a "link without a discernable name" under Web Accessibility.

## What is your proposed solution?
Could we have a field to add a description title, beneath the toggle to 'Link to post'
![Screenshot 2022-05-15 at 19 32 42](https://user-images.githubusercontent.com/1690478/168488447-5e220fbf-c2f9-4cc9-9e86-6e8733da141b.png)
![Screenshot 2022-05-15 at 19 32 49](https://user-images.githubusercontent.com/1690478/168488449-945b4a53-a59a-4647-8a57-43a90b4b4afb.png)

**Comment by fschroiff** [comment]
Current behavior seems to be that the title of the linked post is used as an alt tag. 
Query Block: 
<img width="539" alt="Screenshot 2022-06-02 at 12 14 26" src="https://user-images.githubusercontent.com/660010/171617679-11a10e65-995b-4d35-aa22-3f2c05deb938.png">
One item in the query, the post "Hello World".

Source: 
<img width="653" alt="Screenshot 2022-06-02 at 12 15 11" src="https://user-images.githubusercontent.com/660010/171617715-5f5c3877-5578-4223-93f3-6a73b3b796e5.png">
Image has alt tag

Media Library:
<img width="1359" alt="Screenshot 2022-06-02 at 12 15 42" src="https://user-images.githubusercontent.com/660010/171617745-0de03b13-b707-44b4-89b7-eec623a2dd59.png">
The `alt` tag is set to "LALALA FOOBAR".

Lighthouse:
<img width="705" alt="Screenshot 2022-06-02 at 12 18 20" src="https://user-images.githubusercontent.com/660010/171618240-84c74607-f9d3-484b-867c-f007e86a3f0d.png">
No issues detected.

The title of the post used as an `alt` tag is accessible. An option to add a different / custom title might lead to confusion. 

Tested with Gutenberg 13.3.0 / WP 6.0.

**Comment by Caitboard** [comment]
I think you have a post with an empty title somewhere, I've tried to reproduce this issue and that is the only way I'll get the Axe issue 'No discernible text'. Ensure all of your posts have a title and this shouldn't happen.

**Comment by mirasusanna** [comment]
I also tested this and I don't think anything needs fixing here. I could reproduce the missing link name issue only when the post title was empty. Otherwise the usage of alt texts seems fine to me.

**Comment by carolinan** [comment]
I think that if the featured image is linked but there is no post title, the image alt text should be used, instead of there being an empty alt attribute.

**Comment by carolinan** [comment]
...Though that would require the user to always enter a relevant alt text. It might be better to use the same solution as in the read more block, [where the post ID was used](https://github.com/WordPress/gutenberg/pull/45490/files#diff-5fed9d866f33e17d5bf917f312bcae02af3c8bd244586ab161ed60889c960af6R26).

**Comment by joedolson** [comment]
I believe that this issue should be closed. The changes in https://github.com/WordPress/gutenberg/pull/46830 are merged, and as far as anybody responding to this issue can tell, there are no other circumstances that can create the originally reported problem. I'm closing this, but if you have continuing concerns @blueocto, don't hesitate to follow up!
